src/west/manifest.py

```diff
@@ -855,6 +859,8 @@ def as_dict(self) -> Dict:
         '''
         ret: Dict = {}
         ret['name'] = self.name
+        if self.description:
+            ret['description'] = self.description
```

Comment:
If it's not too much work, could you add a description field to one of the projects in MANIFEST_TEMPLATE in conftest.py, and update test_manifest_freeze() accordingly? Thanks!

Reply:
Of course! I have done this and, as part of it, added better handling of multi-line strings when dumping YAML. I have also described this in the commit message.